### DISCUSSION TITLE
iCub's Range of motion in 3-D space or Cartesian coordinate?

### DISCUSSION THREAD

**Discussion by YANGGANG1971** [robotology/community#428] [Q&A] [answered]
iCub's Range of motion in 3-D space or Cartesian coordinate? (x, y, z)
I want to control the physical iCub, could you give me the technical details?
In the icub wiki, http://wiki.icub.org/wiki/ICub_joints_limit, I only found the joint limit.

**Comment by pattacini** [comment, marked as answer]
Hi @YANGGANG1971 

Unfortunately, we don't have precise details yet regarding the 3D workspace of the iCub, although we deem that this kind of information turns out to be particularly relevant, thus we will certainly publish something in this respect in the near future.

However, I can point you at [`robotology/icub-workspace-estimation`](https://github.com/robotology/icub-workspace-estimation), which is a repository containing some code developed in the past by @alecive right for estimating those quantities you're interested in.

It's not documented but it looks a solid baseline. You could derive pictorial representations of the reaching workspace like the animation reported below (see videos in https://github.com/robotology/icub-workspace-estimation/tree/master/examples):

| iCub Left Arm Workspace |
| :---: |
| ![animation](https://user-images.githubusercontent.com/3738070/92382181-7b5fd100-f10c-11ea-8f3c-09f6dcdf2130.gif) |

Hope this helps for the time being.

**Comment by YANGGANG1971** [comment]
Thanks a lot. It is useful.

**Comment by YANGGANG1971** [comment]
This workspace did not include finger reaching workspace, right? Only for arm?

**Comment by pattacini** [comment]
Usually, the reaching and/or dexterous workspace gets estimated for the end-effector only as it is in this case too.

However, nothing prevents you from applying the same approach to each finger of the hand. In this new context, the workspace would be computed with respect to the frame attached to that hand instead.

**Comment by YANGGANG1971** [comment]
Thanks, I want to ensure in https://github.com/robotology/icub-workspace-estimation/tree/master/examples), whether end-effector refers to the center of the palm.
